test/tools/yulInterpreter/Interpreter.cpp

```diff
@@ -317,14 +320,22 @@ void ExpressionEvaluator::setValue(u256 _value)
 	m_values.emplace_back(std::move(_value));
 }
 
-void ExpressionEvaluator::evaluateArgs(vector<Expression> const& _expr)
+void ExpressionEvaluator::evaluateArgs(
+	vector<Expression> const& _expr,
+	optional<vector<bool>> const& _literalArguments
+)
 {
 	vector<u256> values;
+	size_t i = 0;
 	/// Function arguments are evaluated in reverse.
 	for (auto const& expr: _expr | boost::adaptors::reversed)
 	{
-		visit(expr);
+		if (!_literalArguments || !_literalArguments->at(_expr.size() - i - 1))
+			visit(expr);
+		else
+			m_values = {0};
```

Comment:
`m_values` holds the current value(s) of the expression being evaluated. Why is it zero if the expression contains a literal? Shouldn't it be equal to the value of the literal it contains?

Reply:
That is the actual fix - literal arguments should not be evaluated, but I wanted to put some value there. This is also how the code generator works.